src/Containers/SignUpForm/SignUpForm.tsx

```diff
@@ -1,61 +1,85 @@
-import React from 'react';
-import './SignUpForm.scss';
+import React, { useState, useEffect } from 'react';
+import { UserSignupPosting } from '../../interfaces';
+import InputElement from '../../Components/InputElement/InputElement';
+import { validateCredentials } from '../../_utils';
+
+type checkedInputType = 'first-name' | 'last-name' | 'email' | 'password' | 'repeat-password';
 
 interface Props {
   isLogin: boolean,
   toggleTab: (login: boolean) => void
 }
 
 const SignUpForm: React.FC<Props> = ({ isLogin, toggleTab }) => {
+  const [ user, setUser ] = useState<UserSignupPosting>({
+    email: '',
+    password: '',
+    firstName: '',
+    lastName: '',
+  });
+
+  const [ disabled, setDisabled ] = useState<boolean>(true);
+  const [ repeatPassword, setPassword ] = useState<string>('');
+  const [ error, setError ] = useState<string>('');
+
+  const validateButton = (): void => {
+    if (user.email === '') return setDisabled(true);
+    if (user.password === '') return setDisabled(true);
+    if (user.firstName === '') return setDisabled(true);
+    if (user.lastName === '') return setDisabled(true);
+    if (user.password !== repeatPassword) return setDisabled(true);
+    setError('')
+    return setDisabled(false);
+  }
+
+  const validateInputs = (): void => {
+    if (user.email === '') return setError('Please, add you email');
+    if (user.password === '') return setError('Please, add you password');
+    if (user.firstName === '') return setError('Please, add you first name');
+    if (user.lastName === '') return setError('Please, add you last name');
+    if (user.password !== repeatPassword) return setError('Passwords are not the same');
+    if (!validateCredentials(user.email)) return setError('Please, enter a valid email');
```

Comment:
Working out on this branch and testing some of the functionality, I found that there are never any instances where an error message will populate because of an empty input, so we could trim this method down to the validateCredentials & the password matching.

Reply:
Let's keep some of this logic in place though. We will want to be able to check to see that this email address is already tied to a user account. In case they try to sign up twice

Reply:
Actually, let's keep it but just add it to an onBlur call. And then we can circle back and worry about multiple users, and rendering the appropriate error message.

Reply:
I don't think about deep user validation. There is a lot of stuff that realized on backend side. I left all validation in component for now to get the almost full functionality of component.